src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -479,14 +479,14 @@ public virtual string Href([NotNull] string contentPath)
             return _urlHelper.Content(contentPath);
         }
 
-        private void WritePositionTaggedLiteral(TextWriter writer, string value, int position)
+        private void WritePositionTaggedLiteral([NotNull] TextWriter writer, [NotNull] string value, int position)
         {
             BeginContext(position, value.Length, isLiteral: true);
             WriteLiteralTo(writer, value);
             EndContext();
         }
 
-        private void WritePositionTaggedLiteral(TextWriter writer, PositionTagged<string> value)
+        private void WritePositionTaggedLiteral([NotNull] TextWriter writer, [NotNull] PositionTagged<string> value)
```

Comment:
ditto on comment https://github.com/aspnet/Mvc/pull/2147/files#r26154830 but for both `TextWriter` and `PositionTagged<string>`